etcdctl/ctlv3/command/snapshot_command.go

```diff
@@ -102,8 +102,20 @@ func snapshotSaveCommandFunc(cmd *cobra.Command, args []string) {
 	sp := snapshot.NewV3(lg)
 	cfg := mustClientCfgFromCmd(cmd)
 
+	var (
+		ctx    context.Context
+		cancel context.CancelFunc
+	)
+	if ifCommandTimeoutFlagSet(cmd) {
+		ctx, cancel = commandCtx(cmd)
+	} else {
+		// if user does not specify "--command-timeout" flag, there will be no timeout for snapshot save command
+		ctx, cancel = context.WithCancel(context.Background())
+	}
```

Comment:
seems like you could condense this a bit.
```
ctx, cancel := context.WithCancel(context.Background())
if ifCommandTimeoutFlagSet(cmd) {
	ctx, cancel = commandCtx(cmd)
}
```

Reply:
Thanks!